libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/audio/AudioTypeResolver.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics.audio
+
+import android.content.Context
+import android.media.AudioManager
+import android.os.Build
+import com.mapbox.services.android.navigation.v5.internal.exception.NavigationException
+
+internal sealed class AudioTypeResolver {
+
+    companion object {
+        const val BLUETOOTH = "bluetooth"
+        const val HEADPHONES = "headphones"
+        const val SPEAKER = "speaker"
+        const val UNKNOWN = "unknown"
+    }
+
+    var chain: AudioTypeResolver? = null
```

Comment:
Could we make this non-nullable and `internal`? That way we will be able to remove all the `throw `NavigationException`s in the different `AudioTypeResolver`s. Also `nextChain` is called from `AudioTypeChain` which is `internal` so we can ensure that `AudioTypeResolver` won't be `null` (actually they're non-null 👀 `AudioTypeChain#setup`.